test/sys/test_prctl.rs

```diff
@@ -122,4 +122,40 @@ mod test_prctl {
 
         prctl::set_thp_disable(original).unwrap();
     }
+
+    #[test]
+    fn test_set_addr_name() {
+        use nix::errno::Errno;
+        use nix::sys::mman;
+        use std::ffi::CString;
+        use std::num::NonZeroUsize;
+
+        const ONE_K: libc::size_t = 1024;
+        let sz = NonZeroUsize::new(ONE_K).unwrap();
+        let ptr = unsafe {
+            mman::mmap_anonymous(
+                None,
+                sz,
+                mman::ProtFlags::PROT_READ,
+                mman::MapFlags::MAP_SHARED,
+            )
+            .unwrap()
+        };
+        let err = prctl::set_addr_name(
+            ptr,
+            sz,
+            CStr::from_bytes_with_nul(b"[,$\0").unwrap(),
+        )
+        .unwrap_err();
+        assert_eq!(err, Errno::EINVAL);
+        // `CONFIG_ANON_VMA_NAME` kernel config might not be set
+        prctl::set_addr_name(
+            ptr,
+            sz,
+            CStr::from_bytes_with_nul(b"Nix\0").unwrap(),
+        )
+        .unwrap_or_default();
```

Comment:
Why are we using `.unwrap_or_default()` here

Reply:
because of the comment above, the kernel might not support the feature.

Reply:
Sorry about missing that!

Google search shows that this option is a C macro defined in the kernel code, and if it is not defined, then this prctl call would return  `EINVAL`:

```diff
> +#else /* CONFIG_ANON_VMA_NAME */
> +static int prctl_set_vma(unsigned long opt, unsigned long start,
> +                        unsigned long size, unsigned long arg)
> +{
> +       return -EINVAL;
> +}
> +#endif /* CONFIG_ANON_VMA_NAME */
```

Seems that there is no explicit way to check if this option is set or not?